x/qgb/README.md

```diff
@@ -0,0 +1,143 @@
+# `x/qgb`
+
+## Abstract
+
+This module contains the [Quantum Gravity Bridge](https://blog.celestia.org/celestiums/) (QGB) state machine implementation.
+
+## State machine
+
+The QGB state machine handles the creation of [`AttestationRequestI`](https://github.com/celestiaorg/celestia-app/blob/801a0d412631989ce97748badbd7bb676982db16/x/qgb/types/attestation.go#L15-L23) which can be either a [`DataCommitment`](https://github.com/celestiaorg/celestia-app/blob/801a0d412631989ce97748badbd7bb676982db16/proto/qgb/types.proto#L31-L47) or a [`Valset`](https://github.com/celestiaorg/celestia-app/blob/801a0d412631989ce97748badbd7bb676982db16/proto/qgb/types.proto#L17-L29).
+
+During their creation, the state machine might panic due to an unexpected behavior or event. These panics will be discussed below.
+
+### Conditions of generating a new attestation
+
+#### New valset creation
+
+A new valset is created in the following situations:
+
+```golang
+if (latestValset == nil) || (lastUnbondingHeight == uint64(ctx.BlockHeight())) || significantPowerDiff {
+    ...
+}
+```
+
+- No valset exists in store, so a new valset will be created. This happens mostly after genesis, or after a hard fork.
+- The current block height is the last unbonding height, i.e. when a validator is leaving the validator set. A new valset will need to be created to accommodate that change.
+- A significant power difference happened since the last valset. This could happen if a validator has way more staking power or the opposite. The significant power difference threshold is defined by the constant `SignificantPowerDifferenceThreshold`, and is set to 5% currently.
+
+#### New data commitment creation
+
+A new data commitment is created in the following situation:
+
+```golang
+if ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0 {
+	...
+}
+```
+
+I.e. the current block height is not 0, and we're at a data commitment window height.
+
+The data commitment window is defined as a governance parameter that can be changed. Currently, it is set to 400.
+
+### Panics
+
+During EndBlock step, the state machine generates new attestations if needed. During this generation, the state machine could panic.
+
+#### Data commitment panics
+
+During EndBlock, if the block height corresponds to a `DataCommitmentWindow`, it will generate a new data commitment, during which, the state machine can panic, if it finds invalid state, in the following case:
```

Comment:
```suggestion
During EndBlock, if the block height corresponds to a `DataCommitmentWindow`, it will generate a new data commitment. Note that the state machine can panic if invalid state is introduced.
```

I would also then either remove
`- An unexpected behavior happened while getting the current data commitment:`
or clarify with more precise details of what could cause that. Like if the last nonce's key is missing or something not sure. Also, if we only have a single entry, I'm not sure we need a list

Reply:
I'll remove the data commitment panic section because currently, the `GetDataCommitment` will never return an error => no panics.